apps/backend/src/modules/files/files.service.ts

```diff
@@ -0,0 +1,74 @@
+import { type S3Client } from "@aws-sdk/client-s3";
+
+import { createCommand } from "~/libs/modules/bucket/bucket.js";
+import { config } from "~/libs/modules/config/config.js";
+
+import { FileEntity } from "./files.entity.js";
+import { type FileRepository } from "./files.repository.js";
+
+class FileService {
+	private fileRepository: FileRepository;
+	private s3Client: S3Client;
+
+	public constructor(s3Client: S3Client, fileRepository: FileRepository) {
+		this.s3Client = s3Client;
+		this.fileRepository = fileRepository;
+	}
+
+	public async deleteFile(fileUrl: string): Promise<boolean> {
+		const fileEntity = await this.fileRepository.findByUrl(fileUrl);
+
+		if (!fileEntity) {
+			throw new Error("File not found");
+		}
+
+		const deleteCommand = createCommand({
+			commandType: "delete",
+			params: {
+				Key: fileEntity.toObject().fileKey,
+			},
+		});
+
+		await this.s3Client.send(deleteCommand);
+
+		return await this.fileRepository.delete(fileEntity.toObject().id);
+	}
+
+	public async getFile(fileId: number): Promise<FileEntity | null> {
+		return await this.fileRepository.find(fileId);
+	}
+
+	public async uploadFile({
+		contentType,
+		fileBuffer,
+		fileName,
+	}: {
+		contentType: string;
+		fileBuffer: Buffer;
+		fileName: string;
+	}): Promise<FileEntity> {
+		const fileKey = `${Date.now().toString()}-${fileName}`;
+
+		const uploadCommand = createCommand({
```

Comment:
Shouldn't we create a separate modules for things like that (like mailer or token)

Reply:
@fshabanov should I pass mailer and token into service's constructor like with s3client here?

Reply:
Agree to have a separate s3 module.
@ihorLysak not sure what you mean in your question

Reply:
So do I need to create an s3.module that will handle all operations with the files in the bucket?

Reply:
Yes

Reply:
@fshabanov Maksym here passes the s3client into service's constructor and than uses it via `this.s3client`. If I understand everything correctly, `s3client` is a separate module like `mailer` or `token` (the ones that I implemented). In my code, i just import these modules and use them in methods of the service, while Maksym provides s3client to the service's constructor and than uses it via `this` inside of service's methods.